src/fileActions.jsx

```diff
@@ -467,11 +467,15 @@ const downloadFile = (currentPath, selected) => {
     window.open(`${prefix}?${query}`);
 };
 
-export const fileActions = (path, selected, setSelected, clipboard, setClipboard, addAlert, Dialogs) => {
+export const fileActions = (path, selected, setSelected, clipboard, setClipboard, cwdInfo, addAlert, Dialogs) => {
     const currentPath = path.join("/") + "/";
     const menuItems = [];
 
     const spawnPaste = (sourcePaths, targetPath) => {
+        if (sourcePaths.some(sourcePath => cwdInfo?.entries[basename(sourcePath)])) {
+            addAlert(_("Paste not overwriting existing files"), "danger", "paste-error");
```

Comment:
Not the nicest English, and for many files it'd be good to give a hint -- perhaps "$0 exists, not overwriting with paste". Garrett may have a better idea.